### PR TITLE
Run Apex tests on DartLab

### PR DESCRIPTION
**Pull request by kartheekp-ms**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/1069

Regression? Last working version:

## Description

Thanks to @zivkan for setting up the necessary infrastructure to run functional tests in DartLab. This PR proposes changes required to run Apex tests on DartLab. When running Apex tests, we no longer checkout the host repo and hence we can't execute `build\bootstrap.proj` to download required package. Hence added `nuget.org` as package source when `NetStandard.Library` and `NetCorePlatforms` packages are required to create a project.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] N/A 

- **Documentation**
  
  - [x] N/A

### CODE REVIEW

File: eng/pipelines/templates/pipeline.yml [hunk 1]
```diff
@@ -1,233 +1,240 @@
...
+      - stop
 
 resources:
   pipelines:
-  - pipeline: DartLab
-    source: DartLab
-    branch: main
+    - pipeline: DartLab
+      source: DartLab
+      branch: main
```
Comment (zivkan) [inline eng/pipelines/templates/pipeline.yml]: Why did you change the indenting on the entire file?

Comment (kartheekp-ms) [reply eng/pipelines/templates/pipeline.yml]: I did a format document in VS Code by mistake which indented the entire file. I will revert that change in the next commit.

**Review by zivkan — APPROVED** [review]
🥳

File: eng/pipelines/templates/Apex_Tests_On_Windows.yml [hunk 2]
```diff
@@ -1,115 +1,83 @@
...
+            scriptName: "$(System.DefaultWorkingDirectory)/artifacts/EndToEnd/scripts/SetupFunctionalTests.ps1"
+      postDeployAndRunTestsStepList:
+        - task: PowerShell@1
+          displayName: "Initialize Git Commit Status on GitHub"
+          inputs:
+            scriptType: "inlineScript"
+            inlineScript: |
+              . $(System.DefaultWorkingDirectory)\\artifacts\\EndToEnd\\scripts\\PostGitCommitStatus.ps1
+              SetCommitStatusForTestResult -PersonalAccessToken $(NuGetLurkerPersonalAccessToken) -VstsPersonalAccessToken $(System.AccessToken) -CommitSha $(Build.SourceVersion) -TestName "$(GitHubStatusName)"
+          condition: "not(eq(variables['ManualGitHubChecks'], 'false'))"
```
Comment (dominoFire) [inline eng/pipelines/templates/Apex_Tests_On_Windows.yml:83]: Is it not necessary to run task 'Run Apex Tests (stop on error)' for apex tests anymore ?

File: eng/pipelines/templates/Apex_Tests_On_Windows.yml [hunk 3]
```diff
@@ -1,115 +1,83 @@
...
-    inlineScript: |
-      . $(Build.Repository.LocalPath)\\scripts\\e2etests\\VSUtils.ps1
-      KillRunningInstancesOfVS (Get-LatestVSInstance -VersionRange (Get-VisualStudioVersionRangeFromConfig))
-  condition: "always()"
+        - task: PowerShell@1
+          displayName: "SetupFunctionalTests.ps1"
+          inputs:
+            scriptName: "$(System.DefaultWorkingDirectory)/artifacts/EndToEnd/scripts/SetupFunctionalTests.ps1"
+      postDeployAndRunTestsStepList:
+        - task: PowerShell@1
```
Comment (erdembayar) [inline eng/pipelines/templates/Apex_Tests_On_Windows.yml:76]: Previously there was publish test result step. Is it not necessary anymore?

File: eng/pipelines/templates/Apex_Tests_On_Windows.yml (hunk 2 above)
Comment (dominoFire) [reply eng/pipelines/templates/Apex_Tests_On_Windows.yml:83]: I see on the build Apex tests are executed. So, resolving this comment. Thanks!

**Review by dominoFire — APPROVED** [review]

File: eng/pipelines/templates/Apex_Tests_On_Windows.yml (hunk 3 above)
Comment (kartheekp-ms) [reply eng/pipelines/templates/Apex_Tests_On_Windows.yml:76]: AFAIK, DartLab publishes test results by default and hence I didn't include that as a separate step. For e.g. a test failed once on [CI](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=5264760&view=ms.vss-test-web.build-test-results-tab) for the same private branch and when I navigated to the `Tests` tab, I saw all the test failures along with other information without having an additional step to publish test results.

image

Comment (erdembayar) [reply eng/pipelines/templates/Apex_Tests_On_Windows.yml:76]: I see.  Where is pumping of tests into Dartlab? What is actually instructing Dartlab to run Apex tests?

Comment (kartheekp-ms) [reply eng/pipelines/templates/Apex_Tests_On_Windows.yml:76]: Andy did all the challenging work to inject `NuGet Apex tests` binaries into DartLab drop in the previous PR https://github.com/NuGet/NuGet.Client/blob/dev/eng/pipelines/templates/Build_and_UnitTest.yml#L356 and 
https://github.com/NuGet/NuGet.Client/blob/dev/eng/pipelines/templates/Build_and_UnitTest.yml#L372

Please navigate to `Apex Tests on Windows -> Test Execution -> Run Tests` step which is running the tests using the parameters such as `runSettingsUri` file and apex test binaries passed to it.

**Review by erdembayar — APPROVED** [review]
